frontend/src/components/input-fields/checkbox-input-fields/single-checkbox/single-checkbox-input-field.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+
+import { SingleCheckboxInputFieldProps } from "./single-checkbox-input-field.props.ts";
+import { InputContainer } from "../../frame-input-field";
+import {
+  StyledCheckboxContainer,
+  StyledLabel,
+} from "../checkbox-select-input-field";
+
+export const SingleCheckboxInputField: React.FC<
+  SingleCheckboxInputFieldProps
+> = ({ value: initialValue, text, onChange, ...props }) => {
+  const [isChecked, setisChecked] = useState<boolean>(() => {
+    onChange(initialValue);
+    return initialValue;
+  });
```

Comment:
```suggestion
const [isChecked, setIsChecked] = useState<boolean>(initialValue);

useEffect(() => {
  onChange(initialValue);
}, []);
```

Ich habe mal gelesen, dass es nicht so die feine Art ist, in die use states Funktionen herein zu schreiben, sondern lieber diese Variante verwendet werden sollte.